src/CloneExtensions/ExpressionFactories/ComplexTypeExpressionFactory.cs

```diff
@@ -198,6 +190,29 @@ private IEnumerable<FieldInfo> GetBackingFields(Type type)
             }
         }
 
+        private static IEnumerable<PropertyInfo> GetProperties(Type type)
+        {
+            TypeInfo typeInfo = type.GetTypeInfo();
+
+            while (typeInfo != null && typeInfo.UnderlyingSystemType != _objectType)
+            {
+                var properties = from p in typeInfo.GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                    let setMethod = p.GetSetMethod(false)
+                    let getMethod = p.GetGetMethod(false)
+                    where !p.GetCustomAttributes(typeof(NonClonedAttribute), true).Any()
+                    where setMethod != null && getMethod != null && !p.GetIndexParameters().Any()
+                    where p.DeclaringType == typeInfo.UnderlyingSystemType

```

Comment:
I think you could skip that and add `bindingFlags.DeclaredOnly` when calling `GetProperties`.

Reply:
Done!

Reply:
"I think we also have the problem when a field hides one with the same name from base class" - Marcin 

I do not think so.  Fields can not, apparently, be modified with virtual or abstract key words.  While they can be modified with "new", since we use an array of all public fields both fields are included.  

I added a field to the base class and a "new" field on the sub class to test that it behaves correctly.